src/hooks/UseForm/useForm/useForm.ts

```diff
@@ -0,0 +1,458 @@
+          }),
+          coordinates,
+        )
+
+        return error
+      } else {
+        const typedData = getTypedData<TData, TDimensions>(fieldsGraph.current)
+        const typedFields = {
+          ...typedData.getVertex(coordinates)!,
+          [fieldName]: typedValue,
+        }
+
+        // @ts-expect-error these `TData`s will be the same
+        const error = options?.validate?.(typedValue, typedFields, typedData) ?? null
+
+        updateFieldsVertex(
+          fieldName,
+          field => ({
+            ...field!,
+            error,
+          }),
+          coordinates,
+        )
+
+        return error
+      }
+    },
+    [isRequiredErrorMessageOverride, updateFieldsVertex],
+  )
+
+  const updateFieldsVertexFieldErrors = useCallback(
+    (): void =>
+      fieldsGraph.current.forEachVertex((fields, coordinates) => {
+        if (fields) {
+          Object.keys(fields).forEach(fieldName =>
+            updateFieldsVertexFieldError(fieldName as keyof TData, coordinates),
+          )
+        }
+      }),
+    [updateFieldsVertexFieldError],
+  )
+
+  const updateErrorsVertexFieldError = useCallback(
+    <TFieldName extends keyof TData>(
+      fieldName: TFieldName,
+      coordinates?: CoordinatesOrNever<TDimensions, CoordinatesOfLength<TDimensions>>,
+    ) => {
+      updateErrorsVertex(
+        errors => ({
+          ...errors!,
+          [fieldName]: updateFieldsVertexFieldError(fieldName, coordinates),
+        }),
+        coordinates,
+      )
+    },
+    [updateErrorsVertex, updateFieldsVertexFieldError],
+  )
+
+  const updateErrors = useCallback(() => {
+    updateFieldsVertexFieldErrors()
+
+    const errors = fieldsGraph.current
+      .mapAllVertices(fields =>
+        fields
+          ? Object.keys(fields).reduce(
+              (acc, fieldName) => ({
+                ...acc,
+                [fieldName]: fields[fieldName]?.error,
+              }),
+              {} as Errors<TData>,
+            )
+          : null,
+      )
+      .get()
+
+    setErrors(errors)
+  }, [setErrors, updateFieldsVertexFieldErrors])
+
+  const handleSubmit: HandleSubmit<TData, TDimensions> = useCallback(
+    <TShouldSkipValidations extends boolean = false>({
+      shouldSkipValidations,
+      onSubmit,
+      onError,
+    }: HandleSubmitOptions<TData, TDimensions, TShouldSkipValidations>) => {
+      unregisterInactiveFields()
+
+      if (shouldSkipValidations) {
+        const typedData = getTypedData(fieldsGraph.current)
+        // @ts-expect-error a graph of data is assignable to a graph of partial data
+        onSubmit?.(typedData)
+        resetChanged()
+      } else if (fieldsGraph.current) {
+        updateErrors()
+        const errors = fieldsGraph.current.mapAllVertices(fields =>
+          fields
+            ? Object.keys(fields).reduce(
+                (acc, fieldName) => ({ ...acc, [fieldName]: fields[fieldName]!.error }),
+                {} as Errors<TData>,
+              )
```

Comment:
I did and this only works on these two for the errors because they are partial. Those with required keys still need the cast on the `{}`, so I have just updated these two